File: backend/src/main/scala/microservices/feedback/objects/EscalateFeedbackThreadPlannerRequest.scala
// EscalateFeedbackThreadPlannerRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class EscalateFeedbackThreadPlannerRequest(threadId: String, senderDisplayName: String, body: String)
object EscalateFeedbackThreadPlannerRequest:
  given sourceEncoder: Encoder[EscalateFeedbackThreadPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[EscalateFeedbackThreadPlannerRequest] = deriveDecoder

File: frontend/src/microservices/feedback/objects/EscalateFeedbackThreadPlannerRequest.ts
// 本文件定义 feedback 模块的 `EscalateFeedbackThreadPlannerRequest`，用于把已有线程升级为需要管理员介入的跟进线程。

export type EscalateFeedbackThreadPlannerRequest = {
  senderDisplayName: string
  body: string
}
export const escalateFeedbackThreadRequestFromJson = (json: string): EscalateFeedbackThreadPlannerRequest =>
  JSON.parse(json) as EscalateFeedbackThreadPlannerRequest

export const escalateFeedbackThreadRequestToJson = (value: EscalateFeedbackThreadPlannerRequest): string =>
  JSON.stringify(value)